dnf-json

```diff
@@ -247,6 +247,59 @@ class Solver():
             "dependencies": dependencies
         }
 
+    def chain_depsolve(self, transactions):
+        last_transaction = []
+
+        for idx, transaction in enumerate(transactions):
+            self.base.reset(goal=True)
+            self.base.sack.reset_excludes()
+
+            # don't install weak-deps for transactions after the 1st transaction
+            if idx > 0:
+                self.base.conf.install_weak_deps=False
+
+            # set the packages from the last transaction as installed
+            for installed_pkg in last_transaction:
+                self.base.package_install(installed_pkg, strict=True)
+
+            # depsolve the current transaction
+            self.base.install_specs(
+                transaction.get("package-specs"),
```

Comment:
I think that `get` isn't actually needed on these 3 rows because Go will always marshal these fields (there's no omitempty) so by using `get` we might actually hide a malformed json schema.